redaxo/src/core/lib/util/markdown.php

```diff
@@ -136,14 +136,12 @@ private function handleHeader(array $block = null)
 
         [$level] = sscanf($block['element']['name'], 'h%d');
 
-        if ($level < $this->topLevel || $level > $this->bottomLevel) {
-            return $block;
-        }
+        $displayHeader = $level >= $this->topLevel && $level <= $this->bottomLevel;
 
         if (!isset($block['element']['attributes']['id'])) {
-            $baseId = $id = 'header-'.rex_string::normalize($block['element']['text'], '-');
+            $baseId = $id = rex_string::normalize($block['element']['text'], '-');
```

Comment:
```suggestion
            $baseId = $id = rex_string::normalize($block['element']['text'], '-');
            $baseId = $id = rex_string::normalize(strip_tags($this->{$block['element']['handler']}($block['element']['text'])), '-');
```
Auf die schnelle konnte ich das Problem nur so beheben. Scheint mir aber sowohl unschön als auch ineffizient.

Reply:
@gharlan Hast du hier für eine bessere Idee? Aktuell wird bei Links das Ziel auch in die ID aufgenommen.

Ich habe ein paar Tests gemacht mit der YForm Form Builder Docs Seite: https://docs.google.com/spreadsheets/d/1Y8_cFaV6XTH-n73oR6tMTWz2Uog9kO2URUJ1R1xnSkE/edit?usp=sharing (Eigene Methode ist gleich wie die Methoden "line" und "element" der Parsedown Klasse, aber einfach ohne die Zeilen, welche HTML Tags inkl. Attribute generieren.)

Reply:
> Hast du hier für eine bessere Idee?

Leider noch nicht so recht, schwierig.
Muss mir da die Internals von Parsedown nochmal genauer anschauen, ob es ggf. eine andere geeignete Stelle gibt, wo wir uns reinhängen können. Aber sehe da aktuell leider noch nichts richtig gut passendes.

Reply:
Ok, ich werde mir die Parsedown Klassen im Laufe des Tages auch nochmal genauer anschauen.

Reply:
Ich habe es mir nochmal angeschaut, und keine gute Möglichkeit gefunden. 
Auch wenn es wirklich nicht schön ist, eventuell müssen wir deine obige Variante nehmen.
Zumindest hat man in Überschriften ja meist nicht allzu viel Markup, sodass es sich in Grenzen halten sollte, was wir dann doppelt parsen.

Mir ist übrigens noch aufgefallen, dass wir diesen mit striptags gesäuberten Text dann auch für unsere Sprungnavi nehmen sollten. Aktuell enthalten die Sprunglinks dort die Markdown-Syntax, wenn Überschriften Links o.ä. enthalten.

![Bildschirmfoto 2020-08-05 um 18 43 32](https://user-images.githubusercontent.com/330436/89440324-be183d00-d74b-11ea-94f1-bd10839e072e.png)

Und in den Überschriften haben wir da scheinbar noch ein Styleproblem. Muss aber nicht in diesem PR gelöst werden.

Reply:
Schau aber gerne auch selbst noch mal in Parsedown rein, ob du nicht doch noch einen schöneren Weg findest.

Reply:
Habe eine mögliche Lösung gefunden: https://github.com/aidantwoods/parsedown/blob/4898deca4dbf218784ae70abadb687aeda9f3576/PlainTextParsedownExt.php

Was denkst du? Lohnt sich das einbinden einer zusätzlichen Klasse. -> 833bc70

Reply:
Interessante Idee! :) 

Ich vermute aber, dass wir trotzdem noch striptags benötigen, falls direktes HTML (statt Markdown-Syntax) innerhalb der Überschrift verwendet wurde.
Und ich glaube, ich würde dann doch eher bei deiner vorherigen simplen Variante bleiben, also direkt in den `line()`-Handler innerhalb desselben Objekt einsteigen, und dann anschließend striptags.